File: frontend/src/lib/validations/boosterConfig.ts
import { z } from 'zod';

const boosterItemSchema = z.object({
  unlock_level: z.number().int().min(1, 'Unlock level must be at least 1'),
  refill_amount: z.number().int().min(0, 'Refill amount must be 0 or greater'),
  start: z.number().int().min(0, 'Start amount must be 0 or greater'),
});

export const boosterConfigSchema = z.object({
  undo: boosterItemSchema,
  hint: boosterItemSchema,
  shuffle: boosterItemSchema,
  auto_use_after_ads: z.boolean(),
  time_auto_suggestion: z.number().min(0, 'Time must be 0 or greater'),
  auto_suggestion_enabled: z.boolean(),
});

export type BoosterItem = z.infer<typeof boosterItemSchema>;
export type BoosterConfig = z.infer<typeof boosterConfigSchema>;

const defaultBoosterItem: BoosterItem = {
  unlock_level: 0,
  refill_amount: 0,
  start: 0,
};

export const defaultBoosterConfig: BoosterConfig = {
  undo: { ...defaultBoosterItem },
  hint: { ...defaultBoosterItem },
  shuffle: { ...defaultBoosterItem },
  auto_use_after_ads: false,
  time_auto_suggestion: 0,
  auto_suggestion_enabled: false,
};

File: backend/app/schemas/config_sections/booster_config.py
from pydantic import BaseModel, Field


class BoosterItem(BaseModel):
    """Individual booster configuration"""
    UnlockLevel: int = Field(ge=1, description="Level at which booster unlocks")
    RefillAmount: int = Field(ge=0, description="Amount to refill per interval")
    Start: int = Field(ge=0, description="Starting amount for new players")


class BoosterConfig(BaseModel):
    """Booster configuration for power-ups"""
    Undo: BoosterItem = Field(..., description="Undo booster configuration")
    Hint: BoosterItem = Field(..., description="Hint booster configuration")
    Shuffle: BoosterItem = Field(..., description="Shuffle booster configuration")
    AutoUseAfterAds: bool = Field(default=True, description="Auto-use booster after watching ad")
    
    class Config:
        json_schema_extra = {
            "example": {
                "Undo": {
                    "UnlockLevel": 2,
                    "RefillAmount": 1,
                    "Start": 2
                },
                "Hint": {
                    "UnlockLevel": 4,
                    "RefillAmount": 1,
                    "Start": 2
                },
                "Shuffle": {
                    "UnlockLevel": 3,
                    "RefillAmount": 1,
                    "Start": 2
                },
                "AutoUseAfterAds": True
            }
        }
